src/Idris/IDEMode/CaseSplit.idr

```diff
@@ -65,69 +65,75 @@ isWhitespace : SourcePart -> Bool
+doUpdates defs ups stk (Other ")" :: xs)  = case stk of
+  (Paren :: stk) => pure (Other ")" :: !(doUpdates defs ups stk xs))
+  _ => pure (Other ")" :: !(doUpdates defs ups stk xs))
+doUpdates defs ups stk (Other "[" :: xs)  = pure (Other "[" :: !(doUpdates defs ups (Bracket :: stk) xs))
+doUpdates defs ups stk (Other "]" :: xs)  = case stk of
+  Bracket :: stk => pure (Other "]" :: !(doUpdates defs ups stk xs))
+  stk => pure (Other "]" :: !(doUpdates defs ups stk xs))
+doUpdates defs ups stk (Other "-" :: Other "-" :: xs) = pure (Other "--" :: xs)
+-- name after LBrace
+doUpdates defs ups stk (LBrace :: xs) =
+  let (ws, nws) = span isWhitespace xs in
+    case nws of
+      Name n :: rest => case lookup n ups of
+        Just up => pure (LBrace :: ws ++ Name n :: Whitespace " " :: Equal :: Whitespace " " :: Other up :: !(doUpdates defs ups stk rest))
+        Nothing => pure (LBrace :: !(doUpdates defs ups (Brace :: stk) xs))
+      _ => pure (LBrace :: !(doUpdates defs ups (Brace :: stk) xs))
+-- handle commas directly inside an implicit
+doUpdates defs ups stk (Other "," :: xs) =
+      let (ws, nws) = span isWhitespace xs in
         case nws of
-          -- handle potential whitespace in the other parts
-          Name n :: rest =>
-             let (ws', nws') = span isWhitespace rest in
-               case nws' of
-                  -- brace is immediately closed, so generate a new
-                  -- pattern-match on the values the name can have, e.g.
-                  -- { x}  where x : Nat would become { x = Z}
-                  --                       (and later { x = (S k)})
-                  RBrace :: rest' =>
-                    pure (LBrace :: ws ++
-                          Name n :: Whitespace " " :: Equal :: Whitespace " " ::
-                          !(doUpdates defs ups (Name n :: ws' ++ RBrace :: rest'))
-                          )
-                  -- preserve whitespace before (and after) the Equal
-                  Equal :: rest' =>
-                    let (ws'', nws'') = span isWhitespace rest' in
-                    pure (LBrace :: ws ++
-                          Name n :: ws' ++ Equal :: ws'' ++
-                          !(doUpdates defs ups nws'')
-                          )
-                  -- handle everything else as usual, preserving whitespace
-                  _ => pure (LBrace :: ws ++ Name n :: ws' ++
-                             !(doUpdates defs ups rest)
-                             )
-          -- not a special case: proceed as normal
-          _ => pure (LBrace :: [] ++ !(doUpdates defs ups xs))
+          Name n :: rest => case lookup n ups of
+            Nothing => pure (Other "," :: !(doUpdates defs ups stk xs))
+            Just up => case stk of
+              Brace :: _ => pure (Other "," :: ws ++ Name n :: Whitespace " " :: Equal :: Whitespace " " :: Other up :: !(doUpdates defs ups stk rest))
```

Comment:
What if we have auto?

Reply:
That would hit the default case on line 113, since the `@{ ... }` notation doesn't allow named binding. (It looks like it doesn't allow `,` either, so we'd never hit this case with Auto on the top of the stack.)